app-with-redux-toolkit/src/redux/accountSlice.ts

```diff
@@ -0,0 +1,101 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+
+type InitState = {
+  loading: boolean;
+  amount: number | unknown;
+  error: string;
+};
+
+type ExtraType = {
+  changeAccountState: (count: number, interestRate?: boolean) => number;
+  actualBalance: () => number;
+};
+
+const initialState = {
+  loading: false,
+  amount: 0,
+  error: "",
+} as InitState;
+
+export const depositMoney = createAsyncThunk<
+  number,
+  void,
+  { extra: ExtraType }
+>("depositMoney", async (_, { extra }) => {
+  console.log(extra);
+  return await extra.changeAccountState(1000);
+});
+
+export const withdrawMoney = createAsyncThunk<
+  number,
```

Comment:
tady ty parametry generiku se musí vyplňovat?

v examplu: https://redux-toolkit.js.org/api/createAsyncThunk 

nic takového není, takže bych zbytečný typy smazal, akorát komplikují kód, nebo k čemu tam jsou?

věřím, že ty {extraType} půjdou nějak přidat globálně přes sdílený interface... ale takhle je to strašně moc psaní redundantních typů